packages/tools/lib/create-new-component/index.js

```diff
@@ -108,47 +35,88 @@ const getLibraryName = packageName => {
+
+// Validation of user input
+const isNameValid = name => typeof name === "string" && name.match(/^[a-zA-Z][a-zA-Z0-9_-]*$/);
+
+const generateFiles = (componentName, tagName, library, packageName, isTypeScript) => {
+	componentName = capitalizeFirstLetter(componentName);
+	const filePaths = {
+		"main": isTypeScript 
+			? `./src/${componentName}.ts` 
+			: `./src/${componentName}.js`,
+		"css": `./src/themes/${componentName}.css`,
+		"template": `./src/${componentName}.hbs`,
+	};
+
+	const FileContentTemplate = isTypeScript 
+		? tsFileContentTemplate(componentName, tagName, library, packageName) 
+		: jsFileContentTemplate(componentName, tagName, library, packageName);
+
+	fs.writeFileSync(filePaths.main, FileContentTemplate, { flag: "wx+" });
+	fs.writeFileSync(filePaths.css, "", { flag: "wx+" });
+	fs.writeFileSync(filePaths.template, "<div>Hello World</div>", { flag: "wx+" });
+
+	console.log(`Successfully generated ${filePaths.main}`);
+	console.log(`Successfully generated ${filePaths.css}`);
+	console.log(`Successfully generated ${filePaths.template}`);
+
+	// Change the color of the output
+		console.warn('\x1b[33m%s\x1b[0m', `
+	Make sure to import the component in your bundle by using:
+	import ${componentName} from "./dist/${componentName}.js";`);
 }
 
-const tagName = `ui5-${camelToKebabCase(componentName)}`;
-
-const filePaths = {
-	"js": `./src/${componentName}.js`,
-	"css": `./src/themes/${componentName}.css`,
-	"hbs": `./src/${componentName}.hbs`,
-};
-const sJsFileContentTemplate = jsFileContentTemplate(componentName);
-
-fs.writeFileSync(filePaths.js, sJsFileContentTemplate, { flag: "wx+" });
-fs.writeFileSync(filePaths.css, "", { flag: "wx+" });
-fs.writeFileSync(filePaths.hbs, "<div>Hello World</div>", { flag: "wx+" });
-
+// Main function
+const createWebComponent = async () => {
+	const consoleArguments = process.argv.slice(2);
+	let componentName = consoleArguments[0];
+
+	if (!componentName) {
+		const response = await prompts({
+			type: "text",
+			name: "componentName",
+			message: "Please enter a component name:",
+			validate: (value) => isNameValid(value),
+		});
+		componentName = response.componentName;
+	}
 
-console.log(`Successfully generated ${componentName}.js`);
-console.log(`Successfully generated ${componentName}.css`);
-console.log(`Successfully generated ${componentName}.hbs`);
+	if (consoleArguments.length === 2) {
+		let language = consoleArguments[1];
+		isTypeScript = language === "typescript";
+	} else {
+		const response = await prompts({
+		type: "select",
+		name: "language",
+		message: "Component type:",
+		choices: [
+			{
+				title: "TypeScript (recommended)",
+				value: true,
+			},
+			{
+				title: "JavaScript",
+				value: false,
+			},
+		],
+			initial: 0,
+		});
+		isTypeScript = response.language;
+	}
 
-const bundleLogger = fs.createWriteStream("./bundle.common.js", {
-	flags: "a" // appending
-});
+	const tagName = `ui5-${camelToKebabCase(componentName)}`;
```

Comment:
indeed, it's `ui5- ` for us, but some custom components might like to be called sap-*, or whatever, so I suggest we no longer force the `ui5-` prefix, what do you think @vladitasev

Reply:
I agree, IMO let's have the user select the tag name as a step in the wizard/third parameter of the command line